Задания с 28.09.2021/28.09.2021-Task3/28.09.2021-Task3/Phonebook.c

```diff
@@ -0,0 +1,273 @@
+    char name[50] = { 0 };
+    findName("+7??????????", name, phoneBook, length);
+    return strcmp(name, "RazbeR") == 0;
+}
+
+bool generalTestPassed(void)
+{
+    return saveDataTestPassed() && findPhoneNumberTestPassed() && findNameTestPassed() && addNoteTestPassed();
+}
+
+int main(void)
+{
+    clearFile("Test Database.txt");
+    if (!generalTestPassed())
+    {
+        printf("Tests Failed ...\n");
+        return 1;
+    }
+
+    setlocale(LC_ALL, "Russian");
+    Note phoneBook[100] = { 0 };
+    int notesCount = initializeMemory(phoneBook, "Database.txt");
+    int notesCountBeforeSaving = notesCount;
+
+    while (true)
+    {
+        printf("Введите одну из следующих команд:\n");
+        printf("\n");
+        printf("0 - Выйти\n");
+        printf("1 - Добавить новую запись\n");
+        printf("2 - Распечатать все имеющиеся записи\n");
+        printf("3 - Найти телефон по имени\n");
+        printf("4 - Найти имя по телефону\n");
+        printf("5 - Сохранить текущие данные в файл\n");
+
+        int selectedCommand = 0;
+        const int commandInput = scanf("%d", &selectedCommand);
+        while (getchar() != '\n');
+        if (commandInput != 1)
+        {
+            printf("Кажется, вы ввели недопустимые данные, поэтому программа завершает работу ...\n");
+            return 1;
+        }
+
+        if (selectedCommand == 0)
+        {
+            return 0;
+        }
+        else if (selectedCommand == 1)
+        {
+            if (notesCount == 100)
+            {
+                printf("Телефонный справочник переполнен. Больше нельзя добавлять записи.\n");
+            }
+            else
+            {
+                printf("Введите имя пользователя на английском языке:\n");
+                char name[50] = { 0 };
+                readLineWithoutLineBreak(name, 50);
+                printf("Введите номер телефона:\n");
+                char phoneNumber[30] = { 0 };
+                readLineWithoutLineBreak(phoneNumber, 30);
+
+                addNote(name, phoneNumber, phoneBook, notesCountBeforeSaving);
+                notesCountBeforeSaving++;
+                printf("Запись добавлена\n");
+            }
+        }
+        else if (selectedCommand == 2)
+        {
+            if (notesCountBeforeSaving == 0)
+            {
+                printf("В справочнике пока нет записей, но вы можете их добавить\n");
+            }
+            else
+            {
+                printf("Все имеющиеся в справочнике записи:\n");
+                printAllNotes(phoneBook, notesCountBeforeSaving);
+            }
+        }
+        else if (selectedCommand == 3)
+        {
+            printf("Введите имя пользователя:\n");
+            char name[50] = { 0 };
+            readLineWithoutLineBreak(name, 50);
+            char phoneNumber[30] = { 0 };
+            findPhoneNumber(name, phoneNumber, phoneBook, notesCountBeforeSaving);
+            if (phoneNumber[0] == 0)
```

Comment:
```suggestion
            if (phoneNumber[0] == '\0')
```
Это идеологически правильнее.